systemctl_commands.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+import argparse
+import paramiko
+
+from argparse import RawTextHelpFormatter
+
+
+def build_command(command, service):
+    return "systemctl {} {}".format(command, service)
+
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=RawTextHelpFormatter)
+    parser.add_argument(
+        "-H",
+        "--hostname",
+        dest="hostname",
+        help="Hostname of Client",
+        type=str,
+    )
+    parser.add_argument(
+        "-U",
+        "--user",
+        dest="user",
+        help="remote user to use",
+        type=str,
+        default="root"
+    )
+    parser.add_argument(
+        "-P",
+        "--password",
+        dest="password",
+        help="password for vSphere client",
+        type=str,
+        default=""
+    )
+    parser.add_argument(
+        "-s",
+        "--service",
+        dest="service",
+        help="service to perform command on",
+        type=str
+    )
+    parser.add_argument(
+        "-c",
+        "--command",
+        dest="command",
+        help="command to perform (e.g. stop/start/status/restart)",
+        type=str,
+        default="status",
+    )
+    parser.add_argument(
+        "-p",
+        "--port",
+        dest="port",
+        help="port for the connection",
+        type=int,
+        default=22
+    )
+    parser.add_argument(
+        "-k",
+        "--pubkey",
+        dest="pubkey",
+        help="path to public key file",
+        type=str,
+        default="~/.ssh/id_rsa.pub"
+    )
+    args = parser.parse_args()
+
+    session = paramiko.SSHClient()
+    session.load_system_host_keys()
+    if args.password:
+        session.connect(args.hostname, args.port, args.user, args.password)
+    else:
+        session.connect(args.hostname, args.port, args.user, key_filename=args.pubkey)
```

Comment:
If the key does not exists or fails to authenticate, what error would it throw? if the error message is not clear on itself, maybe we should catch the exception and provide description of failure.

Reply:
Yes, I agree it would be good to introduce some `try/except` methods here.